### ISSUE TITLE
Reduce/remove waiting time for pluggable WiFi devices on startup

### ISSUE THREAD

**Issue by alex3kov** [joukewitteveen/netctl#161]
I use USB WiFi dongle, which has netctl connection profiles. `netctl-auto` is enabled with `systemctl` and when the system boots without the dongle plugged, I'm getting a 1m30s waiting pause for device:
```
Oct 14 08:46:18 arch systemd[1]: sys-subsystem-net-devices-wlp0s20u1.device: Job sys-subsystem-net-devices-wlp0s20u1.device/start timed out.
Oct 14 08:46:18 arch systemd[1]: Timed out waiting for device sys-subsystem-net-devices-wlp0s20u1.device.
Oct 14 08:46:18 arch systemd[1]: Dependency failed for Automatic wireless network connection using netctl profiles.
Oct 14 08:46:18 arch systemd[1]: netctl-auto@wlp0s20u1.service: Job netctl-auto@wlp0s20u1.service/start failed with result 'dependency'.
Oct 14 08:46:18 arch systemd[1]: sys-subsystem-net-devices-wlp0s20u1.device: Job sys-subsystem-net-devices-wlp0s20u1.device/start failed with result 'timeout'.                                
```
Looked for sys-subsystem-net-devices-wlp0s20u1.device, but it doesn't exist even when dongle is plugged in (apparently systemd doesn't create the file).  
Can this 1m30s timeout be disabled?

**Comment by joukewitteveen** [comment]
You could add a file
`/etc/systemd/system/netctl-auto@wlp0s20u1.service.d/timeout.conf` with contents
```TimeoutStartSec=5s```
to reduce the timeout to 5 seconds (see `man systemd.service`). However, this is a workaround. A proper solution is to link the service to the device lifetime. Currently, this is done via `udev` rules. If you can think of a way to interface such rules via `netctl`, I would be happy to know.

**Comment by alex3kov** [comment]
> You could add a file /etc/systemd/system/netctl-auto@wlp0s20u1.service.d/timeout.conf with contents TimeoutStartSec=5s  

I will try this.

> Currently, this is done via udev rules. If you can think of a way to interface such rules via netctl, I would be happy to know.  

So we're back to the same issue as in https://github.com/joukewitteveen/netctl/issues/159. I can't think of a way that wouldn't add to `netctl` a whole subsystem for adding, removing, parsing and keeping track of `udev` rules. This is a bit much for `netctl` IMO. Let's wait and see if there's anything new in https://github.com/systemd/systemd/issues/10406, if not - setting the timeout to 5 sec by default seems reasonable enough.

**Comment by alex3kov** [comment]
> You could add a file /etc/systemd/system/netctl-auto@wlp0s20u1.service.d/timeout.conf with contents TimeoutStartSec=5s to reduce the timeout to 5 seconds (see man systemd.service).

That doesn't work actually - looks like `systemd` is timing out on device itself, not the netctl-auto@... service:
```
A start job is running for sys-subsystem-net-devices-wlp0s20u1.device (50s / 1m30s)
```
I couldn't find how to lower this device timeout in systemd - all the info I could find is about mounting partitions. Tried dropping timeout.conf into /etc/systemd/system/sys-subsystem-net-devices-wlp0s20u1.device.d/, timeout is still 1m30s.

**Comment by alex3kov** [comment]
I'm not sure how to narrow it down and formulate it to `systemd` folks, I doubt they will accept a netctl-specific report. Can you help? Or does last solution in #159 solve this issue as well?

**Comment by joukewitteveen** [comment]
This would become a non-issue for `netctl` once #159 is closed. Whether or not there truly is a bug with `systemd`'s handling of timeouts on device units is not at all netctl-specific.
However, from our perspective, there is no reason to go on a bug-hunt for this.

**Comment by anonymous683854643863** [comment]
@alex3kov I've stumbled upon the same issue recently. I don't know if you're still using netctl, but I've searched the web and found no solutions, so I am posting my own here in hope that it will at least help someone in the future.

I solved the issue by using a udev rule that starts netctl when the device gets registered:
`#/etc/udev/rules.d/20-eth0.rules`
`ACTION=="move", SUBSYSTEM=="net", KERNEL=="net0", ENV{SYSTEMD_WANTS}+="netctl@net0.service"`

The only difference here is that I am dealing with a wired connection, so I'm using netctl and not netctl-auto. You might have to adapt that part to fit your use case.

P.S. On the off chance that you are using old-style device names, you might have to replace "move" with "add" in the rule; see https://github.com/systemd/systemd/issues/2891#issuecomment-753307213 for an explanation. The "predictable" names (like your wlp0s20u1), as well as custom-defined names (like my net0) both require listening to the "move" event.

**Comment by guysoft** [comment]
Perhaps using ``ConditionPathExists=/sys/class/net/%i``? (Suggestion from Debian systemd matrix channel by iam_tj)

**Comment by joukewitteveen** [comment]
> Perhaps using `ConditionPathExists=/sys/class/net/%i`? (Suggestion from Debian systemd matrix channel by iam_tj)

Sure, that could be used in a solution, but I believe the current issue was solved back in [`netctl 1.19`](https://github.com/joukewitteveen/netctl/commit/d511899a3763c07420aa91b106354c8265c6e989), wasn't it? If you last enabled the `netctl-auto` service before installing version 1.19, you may need to re-enable it to benefit from the solution.
